js/glsl/scatter-vertex.glsl

```diff
@@ -50,7 +50,31 @@ attribute vec3 color_previous;
 void main(void) {
     vec3 origin = vec3(xlim.x, ylim.x, zlim.x);
     vec3 size_viewport = vec3(xlim.y, ylim.y, zlim.y) - origin;
+#ifdef USE_SPRITE
+    vec3 vector = v;
+    vec3 vector_previous = v_previous;
+    vec3 position_offset = vec3(x, y, z);
+    vec3 position_offset_previous = vec3(x_previous, y_previous, z_previous);
 
+    float s = mix(size_previous/100., size/100., animation_time_size);
+    vec3 pos = (mix(position_offset_previous, position_offset, vec3(animation_time_x, animation_time_y, animation_time_z))
+                - origin) / size_viewport - 0.5;
+    vec4 posxy = projectionMatrix *
+                modelViewMatrix *
+                vec4(pos,1.0);
+    // Test vector used to correct scale in clipspace
+    vec4 posxy_test = projectionMatrix *
+                modelViewMatrix *
+                vec4(1.0,0.0,0.0,1.0);
+    float posxy_mag = sqrt(posxy_test.x*posxy_test.x +
+                           posxy_test.y*posxy_test.y +
+                           posxy_test.z*posxy_test.z);   
+    s = s * posxy_mag; 
+    gl_Position = posxy + vec4((position.x-0.5)*s, (position.y-0.5)*s, 0, 0);
```

Comment:
position is not set, only for the line, and I think this code can probably share a lot with the #ifndef AS_LINE block.

Reply:
ignore this, it is begin used!

Reply:
Also, I considered sharing a lot of the code with `AS_LINE` block, but there was enough different code, and this ended up being considerably more readable.